src/SchemaParser.f90

```diff
@@ -0,0 +1,1551 @@
+
+      !Parse string into STRa1
+      ALLOCATE(entry(nFields(entryStr)))
+      DO ientry=1,nFields(entryStr)
+        CALL getField(ientry,entryStr,entry(ientry),ierr)
+      ENDDO
+
+      !Add to parameter list
+      CALL paramList%add(TRIM(pListPath),entry)
+
+    ENDSUBROUTINE parseSTRa1_SchEnt
+!
+!-------------------------------------------------------------------------------
+!> @brief Finds the index which corresponds to the provided element name
+!> @param elements  the element array to search for the given name
+!> @param name      the element name we are looking to match in elements
+!> @param index     the index number of the matching element in the array
+!>
+    FUNCTION findElementByName(elements,name) RESULT(index)
+      CLASS(SchemaElementType),INTENT(IN) :: elements(:)
+      TYPE(StringType),INTENT(IN) :: name
+      INTEGER(SIK) :: index
+
+      INTEGER(SIK) :: ielement
+
+      !Ensure the provided blockName matches a block name on this schema
+      index=UNDEFINED_ELEMENT
+      DO ielement=1,SIZE(elements)
+        IF(elements(ielement)%hasName(name)) THEN
+          index=ielement
+          EXIT
+        ENDIF
+      ENDDO
+    ENDFUNCTION findElementByName
+!
+!-------------------------------------------------------------------------------
+!> @brief Determines if the input text file line is currently at End of the File
+!> @param file   the file currently being read
+!> @param isEOF  logical determining whether or not the file is at End of File
+!>
+    FUNCTION atEndOfFile(file) RESULT(isEOF)
+      TYPE(InputFileType),INTENT(IN) :: file
+      LOGICAL(SBK) :: isEOF
+
+      REQUIRE(file%isOpen())
+      isEOF=file%isEOF() .OR. file%getProbe()==DOT
+    ENDFUNCTION atEndOfFile
+!
+!-------------------------------------------------------------------------------
+!> @brief Determines if the input text file line contains readable content
+!> @param file        the file currently being read
+!> @param hasContent  logical determining whether or not the file has content
+!>
+    FUNCTION atContentLine(file) RESULT(hasContent)
+      TYPE(InputFileType),INTENT(IN) :: file
+      LOGICAL(SBK) :: hasContent
+      TYPE(StringType) :: line
+
+      REQUIRE(file%isOpen())
+      hasContent=file%getProbe()/=BANG .AND. file%getProbe() /= DOT
+    ENDFUNCTION atContentLine
+!
+!-------------------------------------------------------------------------------
+!> @brief Checks to see if entry string is empty
+!> @param string   the string to be checked
+!> @param isValid  logical corresponding to if the entry string is empty
+!>
+    FUNCTION isEmptyEntry(string) RESULT(isValid)
+      CHARACTER(LEN=*),PARAMETER :: myName='isEmptyEntry'
+      TYPE(StringType),INTENT(IN) :: string
+      LOGICAL(SBK) :: isValid
+
+      isValid=.FALSE.
+      IF(.NOT.nFields(string)>0) THEN
+        isValid=.TRUE.
+        CALL eSchemaParser%raiseError(modName//'::'//myName// &
+          ' - Entry string contains no entries!')
```

Comment:
Hmm... it doesn't seem appropriate to throw an error in these `is*Entry` functions.  They sound like they're simple query functions that could be called in any situation, but then throw errors.  If the calling routine is expecting a particular outcome, errors should be thrown there instead of inside these functions.

In this case especially, I think the `isEmpyEntry` is the only one that throws an error upon returning `.TRUE.` instead of when returning `.FALSE.`.

Reply:
I've moved error throws out of the query routines and into the parse routines